crates/iota-framework/packages/iota-system/sources/voting_power.move

```diff
@@ -31,26 +31,46 @@ module iota_system::voting_power {
     const EVotingPowerOverThreshold: u64 = 3;
     const EInvalidVotingPower: u64 = 4;
 
+    const ECommitteeMembersSetCorrupt: u64 = 102;
+
+
     /// Set the voting power of all validators.
     /// Each validator's voting power is initialized using their stake. We then attempt to cap their voting power
     /// at `MAX_VOTING_POWER`. If `MAX_VOTING_POWER` is not a feasible cap, we pick the lowest possible cap.
-    public(package) fun set_voting_power(validators: &mut vector<ValidatorV1>) {
+    public(package) fun set_voting_power_v1(validators: &mut vector<ValidatorV1>) {
         // If threshold_pct is too small, it's possible that even when all validators reach the threshold we still don't
         // have 100%. So we bound the threshold_pct to be always enough to find a solution.
         let threshold = TOTAL_VOTING_POWER.min(
             MAX_VOTING_POWER.max(TOTAL_VOTING_POWER.divide_and_round_up(validators.length())),
         );
-        let (mut info_list, remaining_power) = init_voting_power_info(validators, threshold);
+        let (mut info_list, remaining_power) = init_voting_power_info_v1(validators, threshold);
+        adjust_voting_power(&mut info_list, threshold, remaining_power);
+        update_voting_power_v1(validators, info_list);
+        check_invariants_v1(validators);
+    }
+
+    /// Set the voting power of all validators.
+    /// Each validator's voting power is initialized using their stake. We then attempt to cap their voting power
+    /// at `MAX_VOTING_POWER`. If `MAX_VOTING_POWER` is not a feasible cap, we pick the lowest possible cap.
+    public(package) fun set_voting_power_v2(committee_members: &vector<u64>, active_validators: &mut vector<ValidatorV1>) {
+        // If threshold_pct is too small, it's possible that even when all validators reach the threshold we still don't
+        // have 100%. So we bound the threshold_pct to be always enough to find a solution.
+        let threshold = TOTAL_VOTING_POWER.min(
+            MAX_VOTING_POWER.max(TOTAL_VOTING_POWER.divide_and_round_up(committee_members.length())),
+        );
+        let (mut info_list, remaining_power) = init_voting_power_info_v2(committee_members, active_validators, threshold);
         adjust_voting_power(&mut info_list, threshold, remaining_power);
-        update_voting_power(validators, info_list);
-        check_invariants(validators);
+        update_voting_power_v2(active_validators, info_list);
+        check_invariants_v2(committee_members, active_validators);
     }
 
+
+
     /// Create the initial voting power of each validator, set using their stake, but capped using threshold.
     /// We also perform insertion sort while creating the voting power list, by maintaining the list in
     /// descending order using voting power.
     /// Anything beyond the threshold is added to the remaining_power, which is also returned.
-    fun init_voting_power_info(
+    fun init_voting_power_info_v1(
```

Comment:
same here, why do we need a v1?

Reply:
Used in `set_voting_power_v1` which is then indirectly used during genesis creation.